### ISSUE TITLE
Profiling training code

### ISSUE THREAD

**Issue by haanjack** [Transconnectome/4D_fMRI_Transformer#2]
Hi team,

please profile your application and share the result. That would be helpful for me to determine optimization point of your application.

### NVTX tagging
Beforehand, you need to place some code as follows,
```python
torch.cuda.nvtx.range_push("region name")
...
torch.cuda.nvtx.range_pop()
```
These codes will help you to tag code area in range where you want to analysis. In general, cover the training loop, data loader, loss, forward, backward, etc. Also, these can be nested, so please cover specific code area if you want specify more.

### Ranged Profile
Also, use the range profile to focus on the specific steps of the code. In general, you will need 2 or 3 iteration information. Please refer the following example,
```python
while step < num_step:
    if step > 5:
        torch.cuda.cudart().cudaProfilerStart()
    < training codes >
    if step > 7:
        torch.cuda.cudart().cudaProfilerStop()
```

### Profiling
Use the command as follows,
`nsys profile -t cuda,nvtx,cublas,cudnn -d <delay> -f true -o <output-filename> --capture-range=cudaProfilerApi --capture-range-end=stop-shutdown python <options>`
You can find more example from this [doc](https://gist.github.com/mcarilli/376821aa1a7182dfcf59928a7cde3223).

The command will be there if your container has build based on NGC (NVIDIA's container).

### Review
Then download the generated `*.qdrep` file to your machine. You can view the result using [nsight system](https://developer.nvidia.com/nsight-systems).
If you share that file on slack, I'll review and give the feedback.

**Comment by haanjack** [comment]
Or please check my old version of profiling slide.
https://www.slideshare.net/JaeGeunHan/profiling-deep-learning-network-using-nvidia-nsight-systems
